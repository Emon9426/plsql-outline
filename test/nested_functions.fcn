/* 多层嵌套函数测试 */
CREATE OR REPLACE FUNCTION level1_function(p_input NUMBER) RETURN NUMBER IS
  v_level1 NUMBER := p_input;
  
  FUNCTION level2_function(p_param NUMBER) RETURN NUMBER IS
    v_level2 NUMBER := p_param * 2;
    
    FUNCTION level3_function(p_param NUMBER) RETURN NUMBER IS
      v_level3 NUMBER := p_param + 10;
      
      FUNCTION level4_function(p_param NUMBER) RETURN NUMBER IS
        v_level4 NUMBER;
      BEGIN
        v_level4 := p_param * 3;
        RETURN v_level4;
      END level4_function;
      
    BEGIN
      RETURN level4_function(v_level3);
    END level3_function;
    
  BEGIN
    RETURN level3_function(v_level2);
  END level2_function;
  
BEGIN
  RETURN level2_function(v_level1);
END level1_function;
/
