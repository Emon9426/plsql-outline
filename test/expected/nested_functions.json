{
  "name": "level1_function",
  "type": "FUNCTION",
  "children": [
    {
      "name": "level2_function",
      "type": "FUNCTION",
      "children": [
        {
          "name": "level3_function",
          "type": "FUNCTION",
          "children": [
            {
              "name": "level4_function",
              "type": "FUNCTION",
              "children": [
                {
                  "name": "BEGIN",
                  "type": "BLOCK"
                }
              ]
            }
          ]
        }
      ]
    },
    {
      "name": "BEGIN",
      "type": "BLOCK"
    }
  ]
}
